ddtrace/contrib/unittest/workaround_utils.py

```diff
@@ -0,0 +1,37 @@
+import sys
+import traceback
+import unittest
+
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+
+def is_flexmock_erroneous_add_success(instance: unittest.TextTestResult) -> bool:
+    """Returns True if we should work around flexmock's bug of always calling
+    addSuccess regardless of the success of the original test.
+
+    We double-check that we are, in fact, in flexmock's codebase, just in case.
+
+    See https://github.com/flexmock/flexmock/issues/150
+    """
+    # If flexmock is not loaded, then it can't be an erroneous success
+    if "flexmock" not in sys.modules:
```

Comment:
@emmettbutler , I think you had thoughts about how a different PR ( https://github.com/DataDog/dd-trace-py/pull/8011/files/fd3ffc157a6fcc2ac5c80b6e8d69d6b30c77eb94#r1457458249 , IIRC), so I'm guessing you'll have thoughts here too.

I suppose since we're pulling in `pkg_resources` elsewhere, it wouldn't hurt for me to use it here as well?

Reply:
agreed. see if you can standardize the method of checking an installation within ddtrace.